code/modules/vehicles/spaceship.dm

```diff
@@ -0,0 +1,234 @@
+		if(istype(I))
+			if(check_access(I))
+				authenticated = 1
+			if(ACCESS_MARINE_BRIDGE in I.access)
+				authenticated = 2
+		else
+			I = C.wear_id
+			if(istype(I))
+				if(check_access(I))
+					authenticated = 1
+				if(ACCESS_MARINE_BRIDGE in I.access)
+					authenticated = 2
+	if("logout")
+		authenticated = 0
+
+	if (href_list["UP"])
+		do_orbit_checks("UP")
+		cooldown = TRUE
+		addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 1 MINUTES)
+
+	else if (href_list["DOWN"])
+		do_orbit_checks("DOWN")
+		cooldown = TRUE
+		addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 1 MINUTES)
+
+	else if (href_list["escape"])
+		do_orbit_checks("escape")
+		cooldown = TRUE
+		addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 1 MINUTES)
+		//end the round, xeno minor.
+
+	updateUsrDialog()
+
+
+/obj/machinery/computer/navigation/proc/do_orbit_checks(direction)
+	var/current_orbit = GLOB.orbital_mechanics.current_orbit
+	if(cooldown)
+		return
+
+	if(can_change_orbit(current_orbit, direction))
+		if(direction == "escape")
+			var/message = "The [SSmapping.configs[SHIP_MAP].map_name] is preparing to leave orbit.<br><br>Secure all belongings and prepare for engine ignition."
+			priority_announce(message, title = "Retreat")
+			addtimer(CALLBACK(src, .proc/do_change_orbit, current_orbit, direction), 10 SECONDS)
+
+		var/message = "Prepare for orbital change in 10 seconds.<br><br>Moving [direction] the gravity well.<br><br>Secure all belongings and prepare for engine ignition."
+		priority_announce(message, title = "Orbit Change")
+		addtimer(CALLBACK(src, .proc/do_change_orbit, current_orbit, direction), 10 SECONDS)
+
+/obj/machinery/computer/navigation/proc/can_change_orbit(current_orbit, direction)
+	if(cooldown)
+		to_chat(usr, "The ship is currently recalculating based on previous selection.")
+		return FALSE
+	if(changing_orbit)
+		to_chat(usr, "The ship is currently changing orbit.")
+		return FALSE
+	if(direction == "UP" && current_orbit == ESCAPE_VELOCITY)
+		to_chat(usr, "The ship is already at escape velocity! It is already prepped for the escape jump!")
+		return FALSE
+	if(direction == "DOWN" && current_orbit == SKIM_ATMOSPHERE)
+		to_chat(usr, "WARNING, AUTOMATIC SAFETY ENGAGED. OVERRIDING USER INPUT.")
+		return FALSE
+	if(get_power_amount() <= 500000)
+		return FALSE
+	return TRUE
+
+/obj/machinery/computer/navigation/proc/do_change_orbit(current_orbit, direction)
+
+	//chug that sweet sweet powernet juice, like 80% of total
+	if(powered()) //do we still have power?
+		idle_power_usage = 5000
+		addtimer(VARSET_CALLBACK(src, idle_power_usage, 10), 5 MINUTES)
+	else
+		return
+	changing_orbit = TRUE
+	engine_shudder()
+
+	if(direction == "escape")
+		addtimer(CALLBACK(src, .proc/retreat), 10 MINUTES)
+
+	//the bits at the end of the trip.
+	var/message = "Arriving at new orbital level.<br><br>Prepare for engine ignition and stabilization."
+ 	addtimer(CALLBACK(src, .proc/priority_announce, message, title = "Orbit Change"), 290 SECONDS)
```

Comment:
i dont think you can use named arguments here

Reply:
Probably fixed